js/spinbox.js

```diff
@@ -80,6 +80,9 @@
 			this.switches.speed = 500;
 		}
 
+		this.options.defaultUnit = _isUnitLegal(this.options.defaultUnit, this.options.units) ? this.options.defaultUnit : '';
```

Comment:
I haven't run the code yet, but this looks like it doesn't actually use units, this says, "if options.units is legal, use default units, otherwise use an empty string."